### PR TITLE
[tests-only] Add test scenario to check that guests work when share_folder is used

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
When the core system setting `share_folder` is used, guest users have a problem - see the issue link below.

The problem was fixed in core - see the PR link below.

This guests PR adds test coverage. A scenario is added that sets `share_folder` and then verifies that a guest user can use a folder that is shared with them.

The scenario fails when run with core code before https://github.com/owncloud/core/pull/40864
```
Feature: Guests

  Background:                       # /home/phil/git/owncloud/core/apps-external/guests/tests/acceptance/features/apiGuests/guests.feature:4
    Given using OCS API version "1" # FeatureContext::usingOcsApiVersion()
    And using new dav path          # FeatureContext::usingOldOrNewDavPath()

  @email @skipOnOcV10.11 @skipOnOcV10.12
  Scenario: A guest user can upload files to a folder shared with them when shared_folder is in use                                         # /home/phil/git/owncloud/core/apps-external/guests/tests/acceptance/features/apiGuests/guests.feature:575
    Given the administrator has set the default folder for received shares to "Shares"                                                      # OccContext::theAdministratorHasSetTheDefaultFolderForReceivedSharesTo()
    And user "Alice" has been created with default attributes and small skeleton files                                                      # FeatureContext::userHasBeenCreatedWithDefaultAttributes()
    And the administrator has created guest user "guest1" with email "guest1@example.com"                                                   # GuestsContext::theAdministratorHasCreatedAGuestUser()
    And the HTTP status code should be "201"                                                                                                # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And user "Alice" has created folder "/tmp"                                                                                              # FeatureContext::userHasCreatedFolder()
    And user "Alice" has shared folder "/tmp" with guest user "guest1@example.com"                                                          # GuestsContext::userHasSharedFolderWithGuestUser()
    And guest user "guest1" has registered                                                                                                  # GuestsContext::guestUserHasRegistered()
    When user "guest1@example.com" uploads file "textfile.txt" from the guests test data folder to "/tmp/textfile.txt" using the WebDAV API # GuestsContext::userUploadsFileFromGuestsDataFolderTo()
    Then the HTTP status code should be "201"                                                                                               # FeatureContext::thenTheHTTPStatusCodeShouldBe()
      HTTP status code 409 is not the expected value 201
      Failed asserting that 409 matches expected '201'.
    And as "Alice" file "/tmp/textfile.txt" should exist                                                                                    # FeatureContext::asFileOrFolderShouldExist()

--- Failed scenarios:

    /home/phil/git/owncloud/core/apps-external/guests/tests/acceptance/features/apiGuests/guests.feature:575

1 scenario (1 failed)
12 steps (10 passed, 1 failed, 1 skipped)
0m25.98s (18.76Mb)
runsh: Total 1 scenarios (0 passed, 1 failed)
runsh: Exit code of main run: 1
runsh: Total unexpected failed scenarios throughout the test run:
apiGuests/guests.feature:575
```

When I use core from the PR, it passes.

## Related Issue
https://github.com/owncloud/enterprise/issues/5842

and core PR https://github.com/owncloud/core/pull/40864

## How Has This Been Tested?
CI and locally

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/apiGuests/guests.feature [hunk 1]
```diff
@@ -570,3 +570,16 @@ Feature: Guests
     Then the HTTP status code should be "404"
     # uncomment the line below and remove the line above when issue-551 is fixed
     #Then the HTTP status code should be "403"
+
+  @email @skipOnOcV10.11 @skipOnOcV10.12
+  Scenario: A guest user can upload files to a folder shared with them when share_folder is in use
```
Comment (grgprarup) [inline tests/acceptance/features/apiGuests/guests.feature:575]:
```suggestion
  Scenario: guest user can upload files to a folder shared with them when share_folder is in use
```

File: tests/acceptance/features/apiGuests/guests.feature [hunk 2]
```diff
@@ -570,3 +570,16 @@ Feature: Guests
...
+    Given the administrator has set the default folder for received shares to "Shares"
+    And user "Alice" has been created with default attributes and small skeleton files
```
Comment (grgprarup) [inline tests/acceptance/features/apiGuests/guests.feature:577]:
```suggestion
    And user "Alice" has been created with default attributes and without skeleton files
```
Do we need a skeleton?

**Review by saw-jan — APPROVED** [review]

**Review by amrita-shrestha — APPROVED** [review]
lgtm

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**